### PR TITLE
Add Transparent Hugepage information

### LINKED ISSUE #5242: Update kernel tuning doc to mention disabling transparent hugepages
**Not using template on purpose**

[“Transparent Hugepages”](https://www.kernel.org/doc/Documentation/vm/transhuge.txt) is a Linux kernel feature intended to improve performance by making more efficient use of your processor’s memory-mapping hardware. It is enabled (”enabled=always”) by default in most Linux distributions.

Transparent Hugepages gives some applications a [small performance improvement](https://lwn.net/Articles/423590/) (~ 10% at best, 0-3% more typically), but can cause [significant](https://docs.mongodb.com/manual/tutorial/transparent-huge-pages/) [performance](https://blogs.oracle.com/linux/entry/performance_issues_with_transparent_huge) [problems](https://docs.splunk.com/Documentation/Splunk/6.5.2/ReleaseNotes/SplunkandTHP), or even apparent [memory](https://www.digitalocean.com/company/blog/transparent-huge-pages-and-alternative-memory-allocators/) [leaks](https://github.com/golang/go/issues/8832) at worst.

To avoid these problems, you should set `enabled=madvise` on your servers by running
```
echo madvise | sudo tee /sys/kernel/mm/transparent_hugepage/enabled
```

and setting `transparent_hugepage=madvise` on your kernel command line (e.g. in /etc/default/grub).

This change will allow applications that are optimized for transparent hugepages to obtain the performance benefits, and prevent the associated problems otherwise.

https://blog.nelhage.com/post/transparent-hugepages/

### PR DESCRIPTION
**Pull request by nitisht**
## Description
Add description about Transparent Hugepages support and why/how to set it to `madvise`.

## Motivation and Context
Fixes #5242

## How Has This Been Tested?
Tested the commands locally 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by harshavardhana — DISMISSED** [review]

File: docs/deployment/kernel-tuning/README.md [hunk 1]
```diff
@@ -50,6 +50,14 @@ sysctl -w vm.dirty_background_ratio=1
 sysctl -w vm.dirty_ratio=5
 ```
 
+- *`Transparent Hugepage Support`*: This is a Linux kernel feature intended to improve performance by making more efficient use of processor’s memory-mapping hardware. But this may cause [problems](https://blogs.oracle.com/linux/performance-issues-with-transparent-huge-pages-thp) for non optimized applications. As most Linux distributions set it to `enabled=always` by default, we recommend changing this to `enabled=madvise`. This will allow applications optimized for transparent hugepages to obtain the performance benefits, while preventing the associated problems otherwise.
```
Comment (kannappanr) [inline docs/deployment/kernel-tuning/README.md]: ``` for non optimized applications ``` to ``` for non-optimized applications ```

``` optimized for transparent hugepages to obtain the performance benefits, ``` to ``` optimized for transparent huge pages to obtain the performance benefits ```

File: docs/deployment/kernel-tuning/README.md [hunk 2]
```diff
@@ -50,6 +50,14 @@ sysctl -w vm.dirty_background_ratio=1
...
+
+```sh
+echo madvise | sudo tee /sys/kernel/mm/transparent_hugepage/enabled
+```
+
+Also, set `transparent_hugepage=madvise` on your kernel command line (e.g. in /etc/default/grub).
```
Comment (kannappanr) [inline docs/deployment/kernel-tuning/README.md]: If this is going to make this persistent, you might have to specify that. Do you think this needs to be made the first recommendation?

**Review by kannappanr — CHANGES_REQUESTED** [review]
Please let me know if you have any questions on the comments.

File: docs/deployment/kernel-tuning/README.md (hunk 1 above)
Comment (nitisht) [reply docs/deployment/kernel-tuning/README.md]: > optimized for transparent huge pages to obtain the performance benefits

The feature is referred to as _hugepage_, almost all mentions [here](https://www.kernel.org/doc/Documentation/vm/transhuge.txt) are _hugepage_

File: docs/deployment/kernel-tuning/README.md (hunk 2 above)
Comment (nitisht) [reply docs/deployment/kernel-tuning/README.md]: added about persistence. I think the recommendation order is fine, our source blog post mentioned it in the same order (https://blog.nelhage.com/post/transparent-hugepages/)

File: docs/deployment/kernel-tuning/README.md (hunk 1 above)
Comment (nitisht) [reply docs/deployment/kernel-tuning/README.md]: > for non optimized applications to for non-optimized applications

Done

**Review by kannappanr — APPROVED** [review]
LGTM

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**